wavesrv/pkg/blockstore/blockstore_test.go

```diff
@@ -0,0 +1,913 @@
+	SimpleAssert(t, bytes.Equal(readBuf, readTestBytes), "Correct bytes read")
+	Cleanup(t, ctx)
+}
+
+func AppendSyncWorker(t *testing.T, ctx context.Context, wg *sync.WaitGroup) {
+	defer wg.Done()
+	writeBuf := make([]byte, 1)
+	rand.Read(writeBuf)
+	bytesWritten, err := AppendData(ctx, "test-block-id-sync", "file-1", writeBuf)
+	if err != nil {
+		t.Errorf("Worker append err: %v", err)
+	}
+	SimpleAssert(t, bytesWritten == 1, "Correct bytes written")
+}
+func TestAppendSync(t *testing.T) {
+	InitDBState()
+	var wg sync.WaitGroup
+	numWorkers := 10
+	ctx := context.Background()
+	fileMeta := make(FileMeta)
+	fileMeta["test-descriptor"] = true
+	fileOpts := FileOptsType{MaxSize: int64(5 * units.Gigabyte), Circular: false, IJson: false}
+	err := MakeFile(ctx, "test-block-id-sync", "file-1", fileMeta, fileOpts)
+	if err != nil {
+		t.Fatalf("MakeFile error: %v", err)
+	}
+	FlushCache(ctx)
+	for index := 0; index < numWorkers; index++ {
+		wg.Add(1)
+		go AppendSyncWorker(t, ctx, &wg)
+	}
+	wg.Wait()
+	readBuf := make([]byte, numWorkers)
+	bytesRead, err := ReadAt(ctx, "test-block-id-sync", "file-1", &readBuf, 0)
+	if err != nil {
+		t.Errorf("Read Error: %v", err)
+	}
+	log.Printf("read buf : %v", readBuf)
+	SimpleAssert(t, bytesRead == numWorkers, "Correct bytes read")
+	CleanupName(t, ctx, "test-block-id-sync")
+}
+
+func TestAppendSyncMultiple(t *testing.T) {
+	numTests := 100
+	for index := 0; index < numTests; index++ {
+		TestAppendSync(t)
+		log.Printf("finished test: %v", index)
+	}
+}
+
+func WriteAtSyncWorker(t *testing.T, ctx context.Context, wg *sync.WaitGroup, index int64) {
+	defer wg.Done()
+	writeBuf := make([]byte, 1)
+	rand.Read(writeBuf)
+	bytesWritten, err := WriteAt(ctx, "test-block-id-sync", "file-1", writeBuf, index)
+	if err != nil {
+		t.Errorf("Worker append err: %v", err)
+	}
+	log.Printf("worker bytes written: %v %v", bytesWritten, index)
+	SimpleAssert(t, bytesWritten == 1 || bytesWritten == int(index+1), "Correct bytes written")
+}
+
+func TestWriteAtSync(t *testing.T) {
+	InitDBState()
+	var wg sync.WaitGroup
+	numWorkers := 10
+	ctx := context.Background()
+	fileMeta := make(FileMeta)
+	fileMeta["test-descriptor"] = true
+	fileOpts := FileOptsType{MaxSize: int64(5 * units.Gigabyte), Circular: false, IJson: false}
+	err := MakeFile(ctx, "test-block-id-sync", "file-1", fileMeta, fileOpts)
+	if err != nil {
+		t.Fatalf("MakeFile error: %v", err)
+	}
+	FlushCache(ctx)
+	for index := 0; index < numWorkers; index++ {
+		wg.Add(1)
+		go WriteAtSyncWorker(t, ctx, &wg, int64(index))
+	}
+	wg.Wait()
+	readBuf := make([]byte, numWorkers)
+	bytesRead, err := ReadAt(ctx, "test-block-id-sync", "file-1", &readBuf, 0)
+	if err != nil {
+		t.Errorf("Read Error: %v", err)
+	}
+	log.Printf("read buf : %v", readBuf)
+	SimpleAssert(t, bytesRead == numWorkers, "Correct num bytes read")
+	CleanupName(t, ctx, "test-block-id-sync")
+}
+
+func TestWriteAtSyncMultiple(t *testing.T) {
```

Comment:
Sync test for WriteAt
Hopefully this should be good enough to be fairly sure that our sync assumptions are correct